src/contexts/AuthContext.tsx

```diff
@@ -11,20 +11,42 @@ export type TRole = Role | null | undefined;
 export interface AuthContext {
   isAuthenticated: boolean;
   isAdmin: boolean;
+  login: (client: SpvWalletClientExtended) => void;
+  logout: () => void;
 }
 
 const AuthContext = createContext<AuthContext | null>(null);
 
 export const AuthProvider = ({ children }: { children: React.ReactNode }) => {
-  const { spvWalletClient } = useContext(SpvWalletContext) || {};
-
-  const isAuthenticated = !!spvWalletClient;
-  const isAdmin = isAuthenticated && spvWalletClient?.role === Role.Admin;
-  const contextValue = {
-    isAuthenticated,
-    isAdmin,
-  };
-  return <AuthContext.Provider value={contextValue}>{children}</AuthContext.Provider>;
+  const { spvWalletClient } = useSpvWalletClient();
+  const [isAuthenticated, setIsAuthenticated] = useState<boolean>(false);
+  const [isAdmin, setIsAdmin] = useState<boolean>(false);
+
+  useEffect(() => {
+    setIsAuthenticated(!!spvWalletClient);
+    setIsAdmin(isAuthenticated && spvWalletClient?.role === Role.Admin);
+  }, [spvWalletClient, spvWalletClient?.role]);
+
+  const login = useCallback(
+    (client: SpvWalletClientExtended) => {
+      if (client) {
+        setIsAuthenticated(true);
+
+        if (client?.role === Role.Admin) {
+          setIsAdmin(true);
+        }
+      }
+    },
+    [spvWalletClient],
+  );
+
+  const logout = useCallback(() => {
+    setIsAuthenticated(false);
+
+    setIsAdmin(true);
```

Comment:
On `logout` all values should have been set to `false` - I suupose.

Reply:
You're right. Fixed this in adjustXpub page PR

Reply:
Ok 😄 
However, for the future I would prefer to finish (approve & test a PR) before we go to the next step of implementation.